src/components/Image/Image.tsx

```diff
@@ -1,49 +1,87 @@
+    useImperativeHandle<HTMLImageElement | null, HTMLImageElement | null>(
+      forwardRef,
+      () => ref.current
+    );
 
-  if (!img?.src) {
-    return <ImagePlaceholder className={className} />;
-  }
+    if ((!props.src && !props.isSrcLoading) || isError) {
+      return <ImagePlaceholder className={className} />;
+    }
 
-  const shouldLoad = !ref.current?.complete && isLoading;
+    /**
+     *
+     * Using `ref.current.complete` instead of some internal `isLoading` state to check if image
+     * was already loaded as it maintains state on re-render.
+     *
+     * Were we to use some `isLoading` state, then `Image` would flicker on every render.
+     *
+     * Explanation for the flicker:
+     * 1. Let's assume the image was already loaded and rendered, but the user navigated away from
+     *    the page/component that displayed it.
+     * 2. The user decides to navigate back to the page/component with the loaded image.
+     * 3. As the default `isLoading` state is `true`, the `img` component is hidden and
+     *    `ContentLoader` component is displayed.
+     * 4. As  the image is already loaded and cached, the `img.onLoad` triggers immediately and
+     *    updates `isLoading` to `false` triggering a new render
+     * 5. Component `Image` flickers as it transitions from displaying `ContentLoader` to displaying `img`
+     *
+     */
+    const isLoading = !ref.current?.complete && props.isSrcLoading;
 
-  return (
-    <>
-      {shouldLoad && <ContentLoader className={className} />}
-      {!img.isSrcLoading && (
+    return (
+      <>
+        {isLoading && <ContentLoader className={className} />}
+        {/**
+         *
+         * Setting the logic to add `hidden` class name is necessary on both
+         * `WithLink` wrapper and on the child `img`.
+         *
+         * Reason:
+         * if no `href` was passed, that means only the image would be returned and since
+         * it is returned without a wrapper, we need to apply `hidden` className manually.
+         * Otherwise (if `href` was passed), we need to apply `hidden` to the link component
+         * wrapping the `img`.
+         *
+         */}
         <WithLink
-          className={`${className} ${shouldLoad ? "hidden" : ""}`}
-          href={img.href}
-          title={img.title}
+          className={`${className} ${isLoading ? "hidden" : ""}`}
+          href={props.href}
+          title={props.title}
         >
           <img
             ref={ref}
-            src={img.src}
-            className={`${className} object-contain w-full h-full`}
-            alt=""
-            onLoad={() => setLoading(false)}
+            src={props.src}
+            className={`object-contain ${
+              isLoading ? "hidden" : ""
+            } ${className}`}
+            alt={props.alt || ""}
+            loading={props.loading}
+            onError={(e) => (props.onError ? props.onError(e) : setError(true))}
```

Comment:
@SovereignAndrey @ap-justin this logic makes little sense, but it has a reasoning behind it. The usual way would be to treat `props.onError` as a hook into the error event that is run in addition to any internal error handling, like:

```ts
onError={(e) => {
  if (props.onError) props.onError(e);
  // other internal error logic
  // ...
}}
```

But because I wanted to maintain existing behavior in just one component - `Card`, had to implement `onError` here to be used as a direct replacement for any internal error handling (either use `props.onError` OR use internal error logic).

The existing behavior in question is that `Card` wants to just [append a blue background when an error happens with no placeholder image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Marketplace/Cards/Card.tsx#L39). If we could make it so that the `onError` behavior here is updated to just show the [ImagePlaceholder](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/components/Image/ImagePlaceholder.tsx) on error, i.e. use the [default `Image.onError` behavior](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/components/Image/Image.tsx#L27), it would enable us to update `props.onError` to be treated as a hook into the error event (described above).

This would also enable us to remove `onError` in other places where it's used and update them to use some a fallback image. Currently kept their logic as-is as it's less changes to make (see [Leaderboard > Table > Row](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Leaderboard/Table/Row/index.tsx#L20) and [Admin > Proposal > DiffTable](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Admin/Proposal/Content/Preview/DiffTable.tsx#L54)).

Reply:
@ap-justin 
> I have to assume you took it as a negative thing.

No no, just wanted to clarification on what from that comment you "thumbed up". I guess you obviously agreed to the whole comment. Will create a task to update `Card > Image.onError` behaviour.